### PR TITLE
[FEATURE][MC] Token list with network logo and token name

### PR DESCRIPTION
**Pull request by tommasini**
**Description**
This pull request aims to improve the token list, showing the name of the tokens and the network logo, it removes the default '$0.00' of token balances and it's switching them to a skeleton loader.
It was developed unit tests to the components as well.

This PR also have performance improvement on how we fetch the tokens balances.

**Technical Details**

This PR for core repo aims to update the Tokens name:
https://github.com/MetaMask/core/pull/1127/files#diff-7ada7b099f6bcc6f3d4acd71f94b604636abd22cc556a76dbe7b3e610b1d9233

Draft created for performance improve
https://github.com/MetaMask/core/pull/1128

**Test Steps** (All of them are recorded in the Screenshots/Recording section)
- Detect tokens
- Switch account
- Switch network
- Kill and reopen the app
- Show 0 balance main token wallet
- Navigate to buy token
- Hide token (DAI Stable coin) on mainnet
- Import automatically the token (DAI Stable Coin) hidden by address
    - Address: 0x6B175474E89094C44Da98b954EedeAC495271d0F
- Import manually the token hidden with custom token import
    - Address: 0x6B175474E89094C44Da98b954EedeAC495271d0F
- Navigate to the asset page

Non-recorded but tested
* Add token after the swap
* add token with coin gecko API

**Screenshots/Recordings**
Full behaviour on light mode simulator iPhone 14 pro (iOS16.2)
* https://recordit.co/AntCh1rcaK
Full behaviour on dark mode physical device Huawei p30 pro (Android 11)
* https://recordit.co/H30qk9FcyH

**Test Cases**
User imports tokens via token autocomplete across supported networks: https://recordit.co/fRCLO6haNT
User imports tokens by contract address: https://recordit.co/kntnmF0N5n
User imports token on network A then switches to network B: https://recordit.co/nEedgiJEfp
User imports token via asset watcher: https://recordit.co/iXi6IXEwpH
User performs a swap via a dapp then imports token: http://recordit.co/iXtKpUAe2f 
(On the next both recording, there is only the part of the feature but I updated the app from a past version)
User updates the app to this feature with token detection off: https://recordit.co/SGyPxQ21Bl
User updates the app to this feature with token detection already enabled: https://recordit.co/G2iRCJQEEo
User imports an account to wallet then removes that imported account: https://recordit.co/iQEaWdhIPT

**To be improved**
* When the token list it's big, randomly times the endpoint fails (error 429) and we need to refresh the wallet view to update the token balances, we need a callback for this situation, any ideas are welcome

**Issue**

Progresses #

**Checklist**

* [ ] There is a related GitHub issue
* [x] Tests are included if applicable
* [x] Any added code is fully documented

### CODE REVIEW

File: app/components/UI/SearchTokenAutocomplete/index.tsx [hunk 1]
```diff
@@ -55,7 +55,7 @@ const SearchTokenAutocomplete = ({ navigation }: Props) => {
   const [searchQuery, setSearchQuery] = useState('');
   const [selectedAsset, setSelectedAsset] = useState({});
   const [isSearchFocused, setIsSearchFocused] = useState(false);
-  const { address, symbol, decimals, image } = selectedAsset as any;
+  const { address, symbol, decimals, iconUrl, name } = selectedAsset as any;
```
Comment (sethkfman) [inline app/components/UI/SearchTokenAutocomplete/index.tsx:58]: Why was name added to the asset?

Comment (tommasini) [reply app/components/UI/SearchTokenAutocomplete/index.tsx:58]: Thanks for the review!

When we select the asset from the auto complete asset list, we need the name of it to display on the user address token list state, for that reason we need the name, for when we press submit we `TokensController.addToken` with the name to our token list state

Let me know if this answers to the question 🙏🏼

**Review by sethkfman — APPROVED** [review]
LGTM - please answer and resolve the questions for clarity.

File: app/components/Views/Wallet/index.tsx [hunk 2]
```diff
@@ -271,9 +272,10 @@ const Wallet = ({ navigation }: any) => {
     let assets = tokens;
     if (accounts[selectedAddress]) {
       balance = renderFromWei(accounts[selectedAddress].balance);
+
       assets = [
         {
-          name: 'Ether', // FIXME: use 'Ether' for mainnet only, what should it be for custom networks?
+          name: getTicker(ticker) === 'ETH' ? 'Ethereum' : capitalize(ticker),
```
[the remark this answers is not in the document]
Comment (tommasini) [reply app/components/Views/Wallet/index.tsx:278]: I followed the [design](https://www.figma.com/file/NI7m8KldyGDA1GanSY03fy/Latest-Mobile-IA%2FNAV-Redesign?node-id=6779-171168&t=17Qg9NJgqD96ysXa-0) blinded and it was 'Ethereum' on the token name, but you are right 'Ether' it's the token name! I will change it!

[the remark this answers is not in the document]
Comment (tommasini) [reply app/components/Views/Wallet/index.tsx]: Changed !

**Comment by sethkfman** [comment]
@SamuelSalas can you run the e2e and confirm with QA Pass

**Comment by SamuelSalas** [comment]
Detox test results:
Test Suites: 1 failed, 13 passed, 14 total
Tests:       3 failed, 136 passed, 139 total
Snapshots:   0 total
Time:        1702.919 s

e2e/specs/deeplinks.spec.js

**Comment by SamuelSalas** [comment]
Webdriver.IO Test pass

**Comment by tommasini** [comment]
DO NOT MERGE THIS PR

This PR will be closed in order for this [one](https://github.com/MetaMask/metamask-mobile/pull/6020)

**Pull request closed without merging**